File: okj/szoftverfejleszto/morze/java/Idezet.java
import java.util.*;
import java.util.stream.*;

public class Idezet{
    public final String szerzo;
    public final String uzenet;

    public Idezet(String sor, Map<String, String> abc) {
        var split = sor.split(";");

        this.szerzo = morze2Szoveg(split[0], abc);
        this.uzenet = morze2Szoveg(split[1], abc);
    }

    public static String morze2Szoveg(String uzenet, Map<String, String> abc) {
        return Arrays.stream(uzenet.split("       "))
                     .map(k -> Arrays.stream(k.split("   "))
                                     .map(abc::get)
                                     .collect(Collectors.joining()))
                     .collect(Collectors.joining(" "));
    }
}

File: okj/szoftverfejleszto/morze/c#/Idezet.cs
using System.Collections.Generic;

public class Idezet {

    public readonly string szerzo;
    public readonly string uzenet;
    
    public Idezet(string sor, Dictionary<string, string> abc) {
        var split = sor.Split(';');
                
        this.szerzo = Morze.morze2Szoveg(split[0], abc);
        this.uzenet = Morze.morze2Szoveg(split[1], abc);
    }
}